src/Fixer/Alias/PowToExponentiationFixer.php

```diff
@@ -0,0 +1,216 @@
+        for ($i = 1; $i < $end; ++$i) {
+            if (null === $candidate = $this->find('pow', $tokens, $i, $end)) {
+                break;
+            }
+
+            $i = $candidate[1]; // proceed to openParenthesisIndex
+            $candidates[] = $candidate;
+        }
+
+        return $candidates;
+    }
+
+    /**
+     * @param Tokens         $tokens
+     * @param int            $functionNameIndex
+     * @param int            $openParenthesisIndex
+     * @param int            $closeParenthesisIndex
+     * @param array<int,int> $arguments
+     *
+     * @return int number of tokens added to the collection
+     */
+    private function fixPowToExponentiation(Tokens $tokens, $functionNameIndex, $openParenthesisIndex, $closeParenthesisIndex, array $arguments)
+    {
+        // find the argument separator ',' directly after the last token of the first argument;
+        // replace it with T_POW '**'
+        $tokens->overrideAt(
+            $tokens->getNextTokenOfKind(reset($arguments), array(',')),
+            new Token(array(T_POW, '**'))
+        );
+
+        // clean up the function call tokens prt. I
+        $tokens[$closeParenthesisIndex]->clear();
+
+        $added = 0;
+        // check if the arguments need to be wrapped in parenthesis
+        foreach (array_reverse($arguments, true) as $argumentStartIndex => $argumentEndIndex) {
+            if ($this->isParenthesisNeeded($tokens, $argumentStartIndex, $argumentEndIndex)) {
+                $tokens->insertAt($argumentEndIndex + 1, new Token(')'));
+                $tokens->insertAt($argumentStartIndex, new Token('('));
+                $added += 2;
+            }
+        }
+
+        // clean up the function call tokens prt. II
+        $tokens[$openParenthesisIndex]->clear();
+        $tokens[$functionNameIndex]->clear();
+
+        $prev = $tokens->getPrevMeaningfulToken($functionNameIndex);
+        if ($tokens[$prev]->isGivenKind(T_NS_SEPARATOR)) {
+            $tokens[$prev]->clear();
+        }
+
+        return $added;
+    }
+
+    /**
+     * @param Tokens $tokens
+     * @param int    $argumentStartIndex
+     * @param int    $argumentEndIndex
+     *
+     * @return bool
+     */
+    private function isParenthesisNeeded(Tokens $tokens, $argumentStartIndex, $argumentEndIndex)
+    {
+        static $allowedKinds = array(
+            T_DNUMBER, T_LNUMBER, T_VARIABLE, T_STRING, T_OBJECT_OPERATOR, T_CONSTANT_ENCAPSED_STRING, T_DOUBLE_CAST,
+            T_INT_CAST, T_INC, T_DEC, T_NS_SEPARATOR, T_WHITESPACE, T_DOUBLE_COLON, CT_NAMESPACE_OPERATOR, T_LINE,
+            T_COMMENT, T_DOC_COMMENT,
+        );
+
+        for ($i = $argumentStartIndex; $i <= $argumentEndIndex; ++$i) {
+            if ($tokens[$i]->isGivenKind($allowedKinds) || $tokens[$i]->isEmpty()) {
+                continue;
+            }
+
+            if ($tokens[$i]->equals('(')) {
+                $i = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $i);
```

Comment:
why not ignoring all block types by using `$tokens->detectBlockType()`?

`BLOCK_TYPE_ARRAY_SQUARE_BRACE` is missing..

```php
pow([1, 2, 3, 4][$x], 2);
```